src/transactions/feed/detailContent/JumpstartContent.tsx

```diff
@@ -1,35 +1,168 @@
+import { SerializableTransactionRequest } from 'src/viem/preparedTransactionSerialization'
+import { networkIdToNetwork } from 'src/web3/networkConfig'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { Address, Hash, decodeEventLog, encodeFunctionData } from 'viem'
 
 const TAG = 'JumpstartContent'
 
-function JumpstartContent({ transfer }: { transfer: TokenTransfer }) {
+interface Props {
+  transfer: TokenTransfer
+  showOnBottomSheet?: (title: string, description: string, children: React.ReactNode) => void
+}
+
+async function getRewardDataAndStatus(networkId: NetworkId, transactionHash: Hash) {
+  const jumpstartContractAddress = getJumpstartContractAddress(networkId)
+  const viemClient = publicClient[networkIdToNetwork[networkId]]
+
+  const transactionReceipt = await viemClient.getTransactionReceipt({
+    hash: transactionHash,
+  })
+
+  const { eventName, args } = decodeEventLog({
+    abi: walletJumpstart.abi,
+    data: transactionReceipt.logs[1].data,
+    topics: transactionReceipt.logs[1].topics,
+  }) as { eventName: string; args: any }
+
+  if (!['ERC20Deposited', 'ERC721Deposited'].includes(eventName)) {
+    // Sanity check for the event name
+    throw new Error(`Unexpected event name ${eventName}`)
+  }
+
+  const { beneficiary, index }: { beneficiary: Address; index: number } = args
+
+  Logger.debug('Decoded event', { beneficiary, index })
+
+  const erc20Claim = await viemClient.readContract({
+    address: jumpstartContractAddress as Address,
+    abi: walletJumpstart.abi,
+    functionName: 'erc20Claims',
+    args: [beneficiary, index],
+  })
+
+  // @ts-ignore
+  const claimed: boolean = erc20Claim[3]
+
+  Logger.debug(`Reward deposited in ${transactionHash} ${claimed ? 'was' : 'was NOT'} claimed`)
+
+  return { beneficiary, index: Number(index), claimed }
+}
+
+function JumpstartContent({ transfer, showOnBottomSheet }: Props) {
   const { t } = useTranslation()
-  const transferTokenInfo = useTokenInfo(transfer.amount.tokenId)
 
+  const transferTokenInfo = useTokenInfo(transfer.amount.tokenId)
   const parsedAmount = new BigNumber(transfer.amount.value).abs()
   const token = useTokenInfo(transfer.amount.tokenId)
   const isDeposit = transfer.type === TokenTransactionTypeV2.Sent
 
+  const walletAddress = useSelector(walletAddressSelector)
+  const jumpstartContractAddress = getJumpstartContractAddress(transfer.networkId)
+
+  const fetchEscrowData = useAsync(
+    async () => {
+      if (!isDeposit) {
+        return
+      }
+
+      return await getRewardDataAndStatus(transfer.networkId, transfer.transactionHash as Hash)
+    },
+    [],
+    {
+      onError: (error) => {
+        // TODO: Show error message in the UI
+        Logger.error(TAG, 'Failed to fetch escrow data', error)
+      },
+    }
+  )
+
   if (!token) {
     // should never happen
     Logger.error(TAG, 'Token is undefined')
     return null
   }
 
-  // TODO: Add reclaim button
+  const ClaimedLabel = () => (
```

Comment:
i wonder if you can use the Button component with `disabled` and `touchableStyle` set to the green background? maybe you also need to set the color of the font with `fontStyle` but then it'd ensure that this disabled button won't look different from the enabled / loading buttons?